pallets/foreign-investments/src/tests.rs

```diff
@@ -827,6 +827,69 @@ mod investment {
 		});
 	}
 
+	#[test]
+	fn increase_and_fulfill_and_decrease_and_partial_fulfill_and_partial_increase() {
+		new_test_ext().execute_with(|| {
+			util::base_configuration();
+
+			assert_ok!(ForeignInvestment::increase_foreign_investment(
+				&USER,
+				INVESTMENT_ID,
+				AMOUNT,
+				FOREIGN_CURR
+			));
+
+			util::fulfill_last_swap(Action::Investment, AMOUNT);
+
+			assert_ok!(ForeignInvestment::decrease_foreign_investment(
+				&USER,
+				INVESTMENT_ID,
+				AMOUNT,
+				FOREIGN_CURR
+			));
+
+			util::fulfill_last_swap(Action::Investment, foreign_to_pool(3 * AMOUNT / 4));
+
+			MockDecreaseInvestHook::mock_notify_status_change(|_, msg| {
+				assert_eq!(
+					msg,
+					ExecutedForeignDecreaseInvest {
+						amount_decreased: AMOUNT / 2,
+						foreign_currency: FOREIGN_CURR,
+						amount_remaining: AMOUNT / 2,
```

Comment:
I don't understand why `amount_decreased` is `AMOUNT / 2`.  It should be `AMOUNT`, shouldn't it?

Reply:
This is very tricky and always makes me rethink it:

Supposing `AMOUNT` to 100:

| Action       | Pending to send | Pending to swap | Invested |
|--------      |-----------------|-----------------|----------|
| increase 100 | 0               | 100   (->)      | 0        |
| swap 100     | 0               | 0               | 100      |
| decrease 100 | 0               | 100   (<-)      | 0        |
| swap 100     | 75              | 25    (<-)      | 0        |
| increase 50  | 0               | 25    (->)      | 25       |

Basically, when increasing `50`, 
- The pending decrease swap of `25`, is cancelled and invested.
- `25` from "pending to be send" are no longer sent, and becomes pending to be swap.
- `50` from "pending to be send" are sent, because there is no more pending amount to wait.

Reply:
The simpler way of thinking about this: 

If you increase `AMOUNT`, then decrease `AMOUNT`, and again increase in `AMOUNT / 2`, then, the resulting amount in the system is: `AMOUNT - AMOUNT + AMOUNT / 2 = AMOUNT / 2`

Reply:
> If you increase `AMOUNT`, then decrease `AMOUNT`, and again increase in `AMOUNT / 2`, then, the resulting amount in the system is: `AMOUNT - AMOUNT + AMOUNT / 2 = AMOUNT / 2`

I agree with the remaining amount but not the decreased amount. It should not be `AMOUNT / 2` but `AMOUNT` . Otherwise, the message says: "You have decreased your investment by 50 and 50 remaining investment" which is wrong. It should be 100 decreased and 50 remaining.

Reply:
Ouch! I read you wrongly 😓. Then, yes, I think you're right

Reply:
Solved